test/integration/auth/auth_test.go

```diff
@@ -336,18 +337,35 @@ func addTimeoutFlag(URLString string) string {
 	return u.String()
 }
 
-func getTestRequests(namespace string) []struct {
+type testRequest struct {
 	verb        string
 	URL         string
 	body        string
 	statusCodes map[int]bool // allowed status codes.
-} {
-	requests := []struct {
-		verb        string
-		URL         string
-		body        string
-		statusCodes map[int]bool // Set of expected resp.StatusCode if all goes well.
-	}{
+}
+
+func (r *testRequest) restRequest(c clientset.Interface) (*rest.Request, error) {
+	bodyBytes := bytes.NewReader([]byte(r.body))
+
+	u, err := url.Parse(r.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse URL: %v", err)
+	}
+
+	req := c.CoreV1().RESTClient().Verb(r.verb).Body(bodyBytes).AbsPath(u.Path)
```

Comment:
is better to use client-go instead the vanilla http client? 
it adds a things like retries ...

Reply:
Yeah - I realized that I overcomplicated after I sent this out.

Let me try redoing that to make the diff smaller.

Reply:
Fixed